File: clients/php/test/Auth/GuestAuthContextTest.php
<?php declare(strict_types=1);

namespace Eternaltwin\Test\Auth;

use Eternaltwin\Auth\AuthScope;
use Eternaltwin\Auth\GuestAuthContext;
use Eternaltwin\Test\SerializationTestItem;
use PHPUnit\Framework\TestCase;

final class GuestAuthContextTest extends TestCase {
  /**
   * @dataProvider provideFromJson
   * @param SerializationTestItem $item
   */
  public function testFromJson(SerializationTestItem $item): void {
    $actual = GuestAuthContext::fromJson($item->getJson());
    $this->assertEquals($item->getValue(), $actual);
  }

  public function provideFromJson(): array {
    return SerializationTestItem::fromTestDir(
      "core/auth/guest-auth-context",
      [
        "guest" => new GuestAuthContext(
          AuthScope::Default(),
        ),
      ],
    );
  }
}

File: clients/kotlin/src/test/kotlin/net/eternaltwin/auth/GuestAuthContextTest.kt
package net.eternaltwin.auth;

import net.eternaltwin.SerializationTestItem
import net.eternaltwin.user.*
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.MethodSource
import java.util.stream.Stream
import kotlin.test.assertEquals

class GuestAuthContextTest {
  companion object {
    @JvmStatic
    fun fromJsonString(): Stream<SerializationTestItem<GuestAuthContext>> =
      SerializationTestItem.streamFromTestDir(
        "core/auth/guest-auth-context",
        mapOf(
          "guest" to GuestAuthContext(
            scope = AuthScope.Default,
          ),
        ),
      )
  }

  @ParameterizedTest
  @MethodSource
  fun fromJsonString(item: SerializationTestItem<GuestAuthContext>) {
    val actual = GuestAuthContext.fromJsonString(item.jsonString)
    assertEquals(item.value, actual)
  }
}
